### PR TITLE
Use more chars instead of strings

### PR DESCRIPTION
**Pull request by r00ster91**
And it improves things like `gsub(' ', "")`.

### CODE REVIEW

File: src/compiler/crystal/syntax/lexer.cr [hunk 1]
```diff
@@ -491,21 +491,21 @@ module Crystal
             when '\\'
               case char = next_char
               when 'a'
-                io << "\u{7}"
+                io << '\u{7}'
```
Comment (r00ster91) [inline src/compiler/crystal/syntax/lexer.cr]: Or shouldn't this be rather `io << '\a'`?

Comment (straight-shoota) [reply src/compiler/crystal/syntax/lexer.cr]: yes

File: spec/compiler/crystal/tools/expand_spec.cr [hunk 2]
```diff
@@ -22,7 +22,7 @@ private def run_expand_tool(code)
     end
   end
 
-  code = code.gsub('‸', "")
+  code = code.delete('‸')
```
Comment (wontruefree) [inline spec/compiler/crystal/tools/expand_spec.cr]: I like the use of deletes over gsubs 👍

**Comment by wontruefree** [comment]
where you able to measure any performance increase or is this just to make intent clearer?

**Comment by Sija** [comment]
@wontruefree See benchmark in #5882

**Comment by r00ster91** [comment]
Can we maybe merge this now?

**Comment by bararchy** [comment]
@straight-shoota, @RX14 or @sdogruyol 
it seems to passed @bcardiff and seems overall like a good addition, can we have this merged?

**Comment by r00ster91** [comment]
This is ready for a review I think. @jhass @mverzilli

**Comment by r00ster91** [comment]
@ysbaddaden can we merge this please?

**Comment by jwoertink** [comment]
@RX14 This might be a good one to add in to 0.27.

**Comment by j8r** [comment]
What's blocking for the merge?

**Comment by straight-shoota** [comment]
@r00ster91 Could you rebase on master to make sure it's all good?

**Comment by r00ster91** [comment]
There are a bunch of `GC Warning: Failed to expand heap by 16777216 bytes` warnings in the `test_linux32` CI fail but it seems to be unrelated.

**Comment by Sija** [comment]
Ouch, CI is going rogue again :/

Multiple:
- `GC Warning: Finalization cycle involving 0xf9ff990`
- `GC Warning: Failed to expand heap by 16777216 bytes`

Then:

```
Unhandled exception in spawn: Error writing to socket: Broken pipe (Errno)
  from src/socket.cr:77:13 in 'unbuffered_write'
  from src/io/buffered.cr:123:14 in 'write'
  from src/io.cr:853:5 in 'write_byte'
  from src/io/buffered.cr:156:14 in 'write_byte'
  from src/char.cr:767:9 in 'to_s'
  from src/io.cr:184:5 in '<<'
  from src/http/client/response.cr:63:5 in 'to_io'
  from spec/std/http/client/client_spec.cr:13:7 in '->'
  from src/fiber.cr:255:3 in 'run'
  from src/fiber.cr:72:34 in '->'
  from ???
```

Aaand a timeout, triple strike!

**Comment by r00ster91** [comment]
@Sija Yes, I did. I did exactly this on the branch:
```
git fetch origin
git rebase origin/master
```

**Comment by asterite** [comment]
Suspicious... I'm not sure it's rebased against master. Did you `git pull` in master first?

**Comment by j8r** [comment]
It's isn't rebased. Don't forget to `git push -f`

**Comment by straight-shoota** [comment]
No, this branch is based on c163a85b which is 4 days old and doesn't include #7397.

**Comment by r00ster91** [comment]
Oh, no I didn't do `git pull`. I just fetched and rebased and then force-pushed.

**Pull request merged**